Panic after upgrade to 1.0-beta

After upgrade to test 1.0-beta I got this panic:

```
time=2019-04-01T15:13:28.954478674Z app=trickster caller=trickster/main.go:54 level=info event="application start up" name=trickster version=1.0.0
time=2019-04-01T15:13:28.954590951Z app=trickster caller=cache/memory/memory.go:40 level=info event="memorycache setup"
time=2019-04-01T15:13:28.954636574Z app=trickster caller=routing/registration/registration.go:44 level=info event="Registering Prometheus Route Paths" originName=default upstreamHost=localhost:10902
time=2019-04-01T15:13:28.954685424Z app=trickster caller=util/metrics/metrics.go:184 level=info event="metrics http endpoint starting" address= port=8082
time=2019-04-01T15:13:28.954790933Z app=trickster caller=trickster/main.go:61 level=info event="proxy http endpoint starting" address= port=9090
panic: runtime error: index out of range

goroutine 106 [running]:
github.com/Comcast/trickster/internal/proxy.FetchViaObjectProxyCache(0xc0001261c0, 0xae9360, 0xc000134b60, 0xae4fe0, 0xc000142690, 0xf, 0xae0100, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/Comcast/trickster/internal/proxy/objectproxycache.go:37 +0xad3
github.com/Comcast/trickster/internal/proxy.DeltaProxyCacheRequest.func2(0xc0003ef260, 0xc000458070, 0xc00011a280, 0xae9360, 0xc000134b60, 0xae4fe0, 0xc000142690, 0xc0003e9b50, 0xc00010a270, 0xc0001074d0)
	/go/src/github.com/Comcast/trickster/internal/proxy/deltaproxycache.go:175 +0x106
created by github.com/Comcast/trickster/internal/proxy.DeltaProxyCacheRequest
	/go/src/github.com/Comcast/trickster/internal/proxy/deltaproxycache.go:171 +0x139c
```

Using: tricksterio/trickster:1.0-beta
prometheus: thanos-querier 0.3.2

Tested with default config changing only host of the prometheus.
Previous versions of trickster worked fine

Some queries seems to work and others ends on this panic. I can't see a pattern here

--- a/internal/proxy/deltaproxycache.go
+++ b/internal/proxy/deltaproxycache.go
@@ -38,12 +38,7 @@ func DeltaProxyCacheRequest(r *Request, w http.ResponseWriter, client Client, ca
 	}
 	trq.NormalizeExtent()
 
-	a := ""
-	if h, ok := r.Headers[hnAuthorization]; ok {
-		a = h[0]
-	}
-
-	key := client.DeriveCacheKey(r, a)
+	key := client.DeriveCacheKey(r, r.Headers.Get(hnAuthorization))
 	locks.Acquire(key)
 	defer locks.Release(key)
 

--- a/internal/proxy/deltaproxycache.go
+++ b/internal/proxy/deltaproxycache.go
@@ -36,6 +36,7 @@ func DeltaProxyCacheRequest(r *Request, w http.ResponseWriter, client Client, ca
 		ProxyRequest(r, w)
 		return
 	}
+	r.TimeRangeQuery = trq
 	trq.NormalizeExtent()
 
 	key := client.DeriveCacheKey(r, r.Headers.Get(hnAuthorization))

--- a/internal/proxy/influxdb/key.go
+++ b/internal/proxy/influxdb/key.go
@@ -20,36 +20,6 @@ import (
 
 // DeriveCacheKey ...
 func (c Client) DeriveCacheKey(r *proxy.Request, extra string) string {
-
-	k := r.TemplateURL.Path
 	params := r.TemplateURL.Query()
-
-	if p, ok := params[upDB]; ok {
-		k += p[0]
-	}
-
-	if p, ok := params[upQuery]; ok {
-		k += p[0]
-	}
-
-	// Epoch Precision Param
-	if p, ok := params[upEpoch]; ok {
-		k += "." + p[0]
-	}
-
-	// Username Param
-
-	if p, ok := params["u"]; ok {
-		k += "." + p[0]
-	}
-
-	// Password Param
-	if p, ok := params["p"]; ok {
-		k += "." + p[0]
-	}
-
-	if len(extra) > 0 {
-		k += extra
-	}
-	return md5.Checksum(k)
+	return md5.Checksum(r.TemplateURL.Path + params.Get(upDB) + params.Get(upQuery) + params.Get("u") + params.Get("p") + extra)
 }

--- a/internal/proxy/influxdb/key.go
+++ b/internal/proxy/influxdb/key.go
@@ -14,6 +14,8 @@
 package influxdb
 
 import (
+	"strconv"
+
 	"github.com/Comcast/trickster/internal/proxy"
 	"github.com/Comcast/trickster/internal/util/md5"
 )
@@ -21,5 +23,10 @@ import (
 // DeriveCacheKey ...
 func (c Client) DeriveCacheKey(r *proxy.Request, extra string) string {
 	params := r.TemplateURL.Query()
+
+	if r.TimeRangeQuery != nil && r.TimeRangeQuery.Step > 0 {
+		extra += strconv.FormatInt(r.TimeRangeQuery.Step, 10)
+	}
+
 	return md5.Checksum(r.TemplateURL.Path + params.Get(upDB) + params.Get(upQuery) + params.Get("u") + params.Get("p") + extra)
 }

--- a/internal/proxy/objectproxycache.go
+++ b/internal/proxy/objectproxycache.go
@@ -32,11 +32,7 @@ func ObjectProxyCacheRequest(r *Request, w http.ResponseWriter, client Client, c
 // FetchViaObjectProxyCache Fetches an object from Cache or Origin (on miss), writes the object to the cache, and returns the object to the caller
 func FetchViaObjectProxyCache(r *Request, client Client, cache cache.Cache, ttl int, refresh bool, noLock bool) ([]byte, *http.Response) {
 
-	a := ""
-	if h, ok := r.Headers[hnAuthorization]; ok {
-		a = h[0]
-	}
-	key := client.DeriveCacheKey(r, a)
+	key := client.DeriveCacheKey(r, r.Headers.Get(hnAuthorization))
 
 	if !noLock {
 		locks.Acquire(key)

--- a/internal/proxy/prometheus/key.go
+++ b/internal/proxy/prometheus/key.go
@@ -20,25 +20,6 @@ import (
 
 // DeriveCacheKey calculates a query-specific keyname based on the prometheus query in the user request
 func (c Client) DeriveCacheKey(r *proxy.Request, extra string) string {
-
-	k := r.URL.Path
 	params := r.URL.Query()
-
-	if p, ok := params[upQuery]; ok {
-		k += p[0]
-	}
-
-	if p, ok := params[upStep]; ok {
-		k += p[0]
-	}
-
-	if p, ok := params[upTime]; ok {
-		k += p[0]
-	}
-
-	if len(extra) > 0 {
-		k += extra
-	}
-
-	return md5.Checksum(k)
+	return md5.Checksum(r.URL.Path + params.Get(upQuery) + params.Get(upStep) + params.Get(upTime) + extra)
 }

--- a/internal/proxy/request.go
+++ b/internal/proxy/request.go
@@ -17,19 +17,22 @@ import (
 	"net/http"
 	"net/url"
 	"time"
+
+	"github.com/Comcast/trickster/internal/timeseries"
 )
 
 // Request ...
 type Request struct {
-	OriginName    string
-	OriginType    string
-	HandlerName   string
-	HTTPMethod    string
-	URL           *url.URL
-	TemplateURL   *url.URL
-	Headers       http.Header
-	ClientRequest *http.Request
-	Timeout       time.Duration
+	OriginName     string
+	OriginType     string
+	HandlerName    string
+	HTTPMethod     string
+	URL            *url.URL
+	TemplateURL    *url.URL
+	Headers        http.Header
+	ClientRequest  *http.Request
+	Timeout        time.Duration
+	TimeRangeQuery *timeseries.TimeRangeQuery
 }
 
 // NewRequest returns a new proxy request object that can service the downstream request

